File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/stored/job/exceptions/StoredFileJobException.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.stored.job.exceptions;

import com.lasthopesoftware.bluewater.client.library.items.media.files.stored.repository.StoredFile;

public class StoredFileJobException extends Exception implements IStoredFileJobException {
	private StoredFile storedFile;

	public StoredFileJobException(StoredFile storedFile, Throwable innerException) {
		super(innerException);
		this.storedFile = storedFile;
	}

	@Override
	public StoredFile getStoredFile() {
		return storedFile;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/items/files/job/exceptions/StoredFileJobException.kt
package com.lasthopesoftware.bluewater.client.stored.library.items.files.job.exceptions

import com.lasthopesoftware.bluewater.client.stored.library.items.files.repository.StoredFile

class StoredFileJobException(override val storedFile: StoredFile, innerException: Throwable?) :
    Exception(innerException), IStoredFileJobException
